fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneRecordCursor.java

```diff
@@ -306,14 +206,119 @@ private void performScan() throws IOException {
         if (newTopDocs.scoreDocs.length < limit) {
             exhausted = true;
         }
-        topDocs = newTopDocs;
-        if (topDocs.scoreDocs.length != 0) {
-            searchAfter = topDocs.scoreDocs[topDocs.scoreDocs.length - 1];
+        if (newTopDocs.scoreDocs.length != 0) {
+            searchAfter = newTopDocs.scoreDocs[newTopDocs.scoreDocs.length - 1];
         }
+
+        lookupResults = RecordCursor.fromIterator(executor, Arrays.stream(newTopDocs.scoreDocs).iterator())
```

Comment:
The way this pipelining is done, there won't be overlap between the last few stored fields of one search and the next search. Is that okay?

Reply:
Let's probably just use the current approach for now, considering it is just about the performance when matches count is larger than page size, which happens rarely?